The Movie DB ClientTests/MovieList/LocalDataManagerTest.swift

```diff
@@ -7,8 +7,102 @@
 //
 
 import XCTest
+import CoreData
+import Foundation
 @testable import The_Movie_DB_Client
 
 final class LocalDataManagerTest: XCTestCase {
+    fileprivate var localDataManager: MovieListLocalDataManager!
+    fileprivate var inMemoryManagedObjectContext: NSManagedObjectContext!
+    private var coreDataExpectation: XCTestExpectation!
 
+    override func setUp() {
+        super.setUp()
+        let testResources = MovieListLocalDataManager.testResources
+        localDataManager = testResources.localDataManager
+        inMemoryManagedObjectContext = testResources.managedObjectContext
+        coreDataExpectation = self.expectation(description: "Core data actions")
+    }
+
+    override func tearDown() {
+        super.tearDown()
+    }
+
+    func testSaveMovieResponseUpcoming() throws {
+        let mokedMovieResponse = MovieUpcomingResponse.mocked
+
+        NotificationCenter.default.addObserver(
+            forName: .NSManagedObjectContextObjectsDidChange,
+            object: inMemoryManagedObjectContext,
+            queue: nil) { (notification) in
+                guard let userInfo = notification.userInfo else {
+                    XCTFail()
+                    return
+                }
+
+                guard let inserts = userInfo[NSInsertedObjectsKey] as? Set<NSManagedObject> else {
+                    XCTFail()
+                    return
+                }
+
+                guard let insertedMovie = inserts.first as? Movie else {
+                    XCTFail()
+                    return
+                }
+
+                self.coreDataExpectation.fulfill()
+                XCTAssertEqual(insertedMovie.remoteId!, mokedMovieResponse.results.first!.id!)
```

Comment:
Extends [equatable](https://developer.apple.com/documentation/swift/equatable) so it would be nicer to read the test